lib/DownloadManager.js

```diff
@@ -86,7 +86,7 @@ class DownloadManager {
             var fulfill;
             var promise = new Promise(x => fulfill = x);
             logger.info(`git clone ${url} ${folderPath}`);
-            exec('git', ['clone', '--depth', '1', url, folderPath], {timeout: utils.minutes(2)}, (err, stdout, stderr) => {
+            exec('git', ['clone', '--depth', '1', '--branch', branch, url, folderPath], {timeout: utils.minutes(2)}, (err, stdout, stderr) => {
```

Comment:
this would be a breaking behavior as well.
- Before, we would always clone default branch, no matter its name.
- Now, we would be checking out the `master` branch unless the branch is explicitly passed.

This means all current latexonline links to repositories with a non-master default branch would break.

Instead, can we use the `--branch` argument for cloning only when an explicit `branch` was supplied?

Also, IIRC we can save some time by also using the `--single-branch` argument together with `--branch` for the git command.

Reply:
`--depth 1` implies --single-branch, so this wouldn't be needed. 

You are right, this will change default behaviour if the branch is not master. 

BUT: The default behaviour was broken. The `CompilationRequest` always tried to check the `master` for the current commit.